evalml/data_checks/invalid_targets_data_check.py

```diff
@@ -89,6 +91,8 @@ def validate(self, X, y):
                                                     data_check_name=self.name,
                                                     message_code=DataCheckMessageCode.TARGET_HAS_NULL,
                                                     details={"num_null_rows": num_null_rows, "pct_null_rows": pct_null_rows}).to_dict())
+            results["actions"].append(DataCheckAction(DataCheckActionCode.IMPUTE_COL,
+                                                      details={"column": None, "is_target": True, "impute_strategy": "most_frequent"}).to_dict())
```

Comment:
Ahaha yeah I had chosen `most_frequent` as the default to handle both classification and regression but agree that better things can be done for regression--likely the mean. Since `InvalidTargetDataCheck` already takes in the problem_type, it's easy enough to use that to determine which impute_strategy to use.

Thanks for bringing this up, I think it's simple enough that maybe we might as well make this enhancement here :d